ecllibrary/std/OpenSSL.ecl

```diff
@@ -0,0 +1,329 @@
+     *
+     * @see     IVSize()
+     *          SaltSize()
+     *          Encrypt()
+     *          Decrypt()
+     */
+    EXPORT DATASET({STRING name}) AvailableAlgorithms() := lib_openssl.OpenSSL.cipherAvailableAlgorithms();
+
+    /**
+     * Return the size of the IV used for the given symmetric
+     * cipher algorithm.
+     *
+     * This is primarily an introspection/discovery function. Once
+     * you determine the proper value for the algorithm you want to
+     * use, you should hardcode it.
+     *
+     * @param   algorithm   The name of the symmetric cipher to examine;
+     *                      must be one of the values returned from
+     *                      the AvailableAlgorithms() function in
+     *                      this module; cannot be empty; REQUIRED
+     *
+     * @return  The size of the IV used by the given algorithm, in bytes.
+     *
+     * @see     AvailableAlgorithms()
+     */
+    EXPORT UNSIGNED2 IVSize(VARSTRING algorithm) := lib_openssl.OpenSSL.cipherIVSize(algorithm);
+
+    /**
+     * Return the size of the salt used for the given symmetric
+     * cipher algorithm.
+     *
+     * This is primarily an introspection/discovery function. Once
+     * you determine the proper value for the algorithm you want to
+     * use, you should hardcode it.
+     *
+     * @param   algorithm   The name of the symmetric cipher to examine;
+     *                      must be one of the values returned from
+     *                      the AvailableAlgorithms() function in
+     *                      this module; cannot be empty; REQUIRED
+     *
+     * @return  The size of the salt used by the given algorithm, in bytes.
+     *
+     * @see     AvailableAlgorithms()
+     */
+    EXPORT UNSIGNED2 SaltSize(VARSTRING algorithm) := 8;
+
+    /**
+     * Encrypt some plaintext with the given symmetric cipher and a
+     * passphrase. Optionally, you can specify static IV and salt values.
+     * The encrypted ciphertext is returned as a DATA value.
+     *
+     * If IV or salt values are explicitly provided during encryption then
+     * those same values must be provided during decryption.
+     *
+     * @param   plaintext   The data to encrypt; REQUIRED
+     * @param   algorithm   The name of the symmetric cipher to use;
+     *                      must be one of the values returned from
+     *                      the AvailableAlgorithms() function in
+     *                      this module; cannot be empty; REQUIRED
+     * @param   iv          The IV to use during encryption; if not set
+     *                      then a random value will be generated; if set,
+     *                      it must be of the expected size for the given
+     *                      algorithm; OPTIONAL, defaults to creating a
+     *                      random value
+     * @param   salt        TCURRENT_OPENSSL_VERSIONencryption; if not set
```

Comment:
It looks like a 'replace all' event went wrong here.

Reply:
Fixed.